synapse/handlers/profile.py

```diff
@@ -368,6 +373,43 @@ def set_avatar_url(self, target_user, requester, new_avatar_url, by_admin=False)
                 400, "Avatar URL is too long (max %i)" % (MAX_AVATAR_URL_LEN, ),
             )
 
+        # Enforce a max avatar size if one is defined
+        if (
+            self.media_repo_enabled and
+            (self.max_avatar_size or self.allowed_avatar_mimetypes)
+        ):
+            # Parse the media URI
+            try:
+                media_id = new_avatar_url.split("/")[-1]
```

Comment:
This never actually throws btw, since `.split` will always return at least one thing

Reply:
I think this is what I was looking for with the `ValueError`:

```
            try:
                _, media_id = new_avatar_url.split("/")
            except ValueError:
                raise SynapseError(400, "Invalid avatar URL '%s' supplied" %
                                   new_avatar_url)
```